scripts/analysis/gradle_plugins/gradle_plugins_analysis.py

```diff
@@ -0,0 +1,68 @@
+import argparse
+import os
+import pandas as pd
+import sys
+from typing import Dict, Any
+from utils import Extensions, create_directory
+
+from analysis.gradle_plugins.column_names_utils import GradlePluginsStatsColumn
+
+GradlePluginsStats = Dict[str, Any]
+
+
+def save_stats_with_to_csv(path_to_dir: str, filename: str, stats: GradlePluginsStats):
+    csv_file_path = os.path.join(path_to_dir, filename)
+    pd.DataFrame.from_dict(stats) \
+        .sort_values(by="count", ascending=False) \
+        .to_csv(csv_file_path, index=False)
+
+
+def get_plugins_stats(df: pd.DataFrame, unique=False) -> GradlePluginsStats:
+    plugins = {}
+    projects_by_plugins = {}
+    for project_name, plugin_id in df.values:
+        if plugin_id not in plugins:
```

Comment:
Maybe can you use [defaultdict](https://docs.python.org/3/library/collections.html#collections.defaultdict)?